test/ts/unit/collateralizer/collateralizer.ts

```diff
@@ -0,0 +1,244 @@
+
+    const TX_DEFAULTS = { from: CONTRACT_OWNER, gas: 4000000 };
+
+    before(async () => {
+        tokenRegistry = await TokenRegistryContract.deployed(web3, TX_DEFAULTS);
+        mockDebtRegistry = await MockDebtRegistryContract.deployed(web3, TX_DEFAULTS);
+        mockToken = await MockERC20TokenContract.deployed(web3, TX_DEFAULTS);
+        mockTokenRegistry = await MockTokenRegistryContract.deployed(web3, TX_DEFAULTS);
+        mockTokenTransferProxy = await MockTokenTransferProxyContract.deployed(web3, TX_DEFAULTS);
+        mockTermsContract = await MockCollateralizedTermsContractContract.deployed(web3, TX_DEFAULTS);
+
+        /*
+        In our test environment, we want to interact with the contract being
+        unit tested as a statically-typed entity. In order to accomplish this,
+        we take the following steps:
+
+          1 - Instantiate an instance of the contract through the Truffle
+              framework.
+          2 - Instantiate an instance of the contract through the Web3 API using
+              the truffle instance's ABI.
+          3 - Use the Web3 contract instance to instantiate a statically-typed
+              version of the contract as handled by ABI-GEN, which generates
+              a contract wrapper with types pulled from the contract's ABI.
+         */
+
+        // Step 1: Instantiate a truffle instance of the contract.
+        const collateralContractTruffle = await collateralizer.new(
+            MOCK_DEBT_KERNEL_ADDRESS,
+            mockDebtRegistry.address,
+            mockTokenRegistry.address,
+            mockTokenTransferProxy.address,
+            { from: CONTRACT_OWNER },
+        );
+
+        // Step 2: Instantiate a web3 instance of the contract.
+        const collateralContractWeb3Contract = web3.eth
+            .contract(collateralizer.abi)
+            .at(collateralContractTruffle.address);
+
+        // Step 3: Instantiate a statically-typed version of the contract.
+        collateralizerContract = new CollateralizerContract(
+            collateralContractWeb3Contract,
+            TX_DEFAULTS,
+        );
+
+        const testContracts = {
+            collateralizer: collateralizerContract,
+            mockCollateralToken: mockToken,
+            mockDebtRegistry,
+            mockTokenRegistry,
+            mockTokenTransferProxy,
+            mockTermsContract,
+        };
+
+        const testAccounts = {
+            ATTACKER,
+            BENEFICIARY_1,
+            BENEFICIARY_2,
+            COLLATERALIZER,
+            NON_COLLATERALIZER,
+            MOCK_DEBT_KERNEL_ADDRESS,
+            MOCK_TERMS_CONTRACT_ADDRESS,
+        };
+
+        // Grant the terms contract authorization to call the `collateralize` function.
+        await collateralizerContract.addAuthorizedCollateralizeAgent.sendTransactionAsync(
+            mockTermsContract.address,
+        );
+
+        await collateralizerContract.addAuthorizedCollateralizeAgent.sendTransactionAsync(
```

Comment:
Yeah, it's because I couldn't use `mockTermsContract.address` as a sender when calling `collateralize()` directly - results in a "unrecognized sender account" error. I'll add comments for that, unless you have another suggestion!

Reply:
Yeah, I figured -- I feel like we def need comments, but also could probably change the name of the variable to something a bit more separate so that there's a distinction between the two.  Maybe, `mockTermsContractOne.address` and `MOCK_TERMS_CONTRACT_ADDRESS_TWO`, with a comment explaining the difference between the two?